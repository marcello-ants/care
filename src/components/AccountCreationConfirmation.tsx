import { ReactElement } from 'react';
import { Icon64BrandActionSuccess } from '@care/react-icons';
import { Grid, Typography } from '@material-ui/core';
import { StatelessSelector, Pill } from '@care/react-component-lib';
import { makeStyles } from '@material-ui/core/styles';
import Header from '@/components/Header';
import { SelectorType } from '@care/react-component-lib/dist/components/Selector/StatelessSelector';

const useStyles = makeStyles((theme) => ({
  selector: {
    '& .MuiListItem-root': {
      marginBottom: 0,
    },
  },
  icon: {
    position: 'relative',
    left: '-9px',
  },
  bodyText: {
    fontWeight: 'normal',
    lineHeight: '24px',
    margin: theme.spacing(2, 0, 4),
  },
  typeContainer: {
    margin: theme.spacing(1, 0, 1),
  },

  buttonContainer: {
    padding: theme.spacing(3.5, 3, 0),
  },
  header: {
    marginBottom: ({ subheader }: { subheader: string | undefined }) =>
      subheader ? 0 : theme.spacing(1),
  },
}));

type Options = {
  label: string;
  value: string;
};

interface IAccountCreationConfirmation {
  header: string;
  description?: string;
  subheader?: string;
  options: Options[];
  onChange?: SelectorType['onChange'];
  cta?: ReactElement;
  value?: string;
  showIcon?: boolean;
}

function AccountCreationConfirmation({
  header,
  description,
  subheader,
  options,
  value,
  onChange,
  cta,
  showIcon,
}: IAccountCreationConfirmation) {
  const classes = useStyles({ subheader });

  return (
    <Grid container>
      {showIcon && (
        <Grid item xs={12}>
          {/*
          The svg has a litle padding that makes the icon smaller
          60px makes the icon size to 49.33 the correct size is 60px approx
          74px makes the icon size 60.88
        */}
          <Icon64BrandActionSuccess size="74px" className={classes.icon} />
        </Grid>
      )}

      <Grid item xs={12} className={classes.header}>
        <Header>{header}</Header>
      </Grid>
      {description && (
        <Grid item xs={12}>
          <Typography variant="body2" className={classes.bodyText}>
            {description}
          </Typography>
        </Grid>
      )}

      {subheader && (
        <Grid item xs={12}>
          <Typography variant="h4">{subheader}</Typography>
        </Grid>
      )}

      <Grid item xs={12} className={classes.typeContainer}>
        <StatelessSelector onChange={onChange} single value={value} className={classes.selector}>
          {options.map((option) => (
            <Pill key={option.label} {...option} size="md" />
          ))}
        </StatelessSelector>
        {cta && <div className={classes.buttonContainer}>{cta}</div>}
      </Grid>
    </Grid>
  );
}

AccountCreationConfirmation.defaultProps = {
  value: '',
  onChange: () => {},
  cta: null,
  showIcon: true,
  description: '',
  subheader: '',
};

export default AccountCreationConfirmation;
