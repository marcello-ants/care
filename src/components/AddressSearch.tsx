import { useEffect, useState } from 'react';
import { theme } from '@care/material-ui-theme';
import { Box, Button, makeStyles } from '@material-ui/core';
import { GoogleMaps, GoogleMapsActions, useGoogleMaps } from '@care/seeker-react-components';
import { Icon24UtilitySearch } from '@care/react-icons';
import { InlineTextField, Typography } from '@care/react-component-lib';
import { AddressInput } from '@/__generated__/globalTypes';
import { logIbAddressErrorEvent } from '@/utilities/analyticsPagesUtils';

const useStyles = makeStyles(() => ({
  addressSearchContainer: {
    position: 'relative',
    marginTop: theme.spacing(-2.75),
    '&& > .MuiTextField-root': {
      paddingLeft: 0,
      paddingRight: 0,
    },
    '&& label': {
      paddingLeft: theme.spacing(5),
      textAlign: 'left',
    },
    '&& input': {
      paddingLeft: theme.spacing(7),
    },
  },
  loopIcon: {
    position: 'absolute',
    top: 42,
    left: 20,
    zIndex: 2,
  },
  locationContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  locationText: {
    textAlign: 'left',
    whiteSpace: 'pre-wrap',
  },
  editLocationButton: {
    width: 81,
  },
  notFoundContainer: {
    padding: theme.spacing(5, 0),
    textAlign: 'center',
  },
  suiteContainer: {
    '&& > .MuiTextField-root': {
      paddingLeft: 0,
      paddingRight: 0,
    },
  },
}));

interface AddressSearchProps {
  seekerAddress: AddressInput;
  setSeekerAddress: (address: AddressInput) => void;
  displayAddress: string;
  setDisplayAddress: (displayAddress: string) => void;
  pageName: string;
}

const AddressSearch = ({
  seekerAddress,
  setSeekerAddress,
  displayAddress,
  setDisplayAddress,
  pageName,
}: AddressSearchProps): JSX.Element => {
  const classes = useStyles();
  const [{ placeDetails, predictions }, dispatch] = useGoogleMaps();
  const addressLabelText = 'Street address';
  const addressLabelTextOnInput = 'Search address';
  const suiteLabelText = 'Apt or Suite (optional)';
  const invalidAddressText = 'Invalid address';
  const notFoundText = 'Address not found';
  const notFoundDescription = `Make sure it's spelled correctly or try adding a city, state or zip code`;
  const CITY_POSITION: number = 2;
  const STATE_POSITION: number = 3;

  const [label, setLabel] = useState(addressLabelText);
  const [showAddressSearch, setShowAddressSearch] = useState<boolean>(true);
  const [shouldSaveAddress, setShouldSaveAddress] = useState<boolean>(false);

  const isValidAddress = seekerAddress.zip && seekerAddress.addressLine1;

  const handleFocus = () => {
    setLabel(addressLabelTextOnInput);
  };

  const handleBlur = () => {
    setLabel(addressLabelText);
  };

  const setGoogleMapsPlaceInfo = (inputText: string) => {
    dispatch({ type: GoogleMapsActions.SET_INPUT_SEARCH, payload: inputText });
    dispatch({ type: GoogleMapsActions.SET_PLACE_DETAILS, payload: null });
    dispatch({ type: GoogleMapsActions.CLEAR_PREDICTIONS });
  };

  const handleEditClick = () => {
    setGoogleMapsPlaceInfo(seekerAddress.addressLine1);
    setShowAddressSearch(true);
    setShouldSaveAddress(false);
  };

  const handlePredictionClick = () => {
    setShowAddressSearch(false);
    setShouldSaveAddress(true);
  };

  useEffect(() => {
    if (shouldSaveAddress && placeDetails) {
      setSeekerAddress({
        ...seekerAddress,
        addressLine1: placeDetails.prediction.structured_formatting.main_text,
        city: placeDetails.prediction.terms[CITY_POSITION].value,
        state: placeDetails.prediction.terms[STATE_POSITION].value,
        zip: placeDetails.zip || '',
        longitude: placeDetails.longitude,
        latitude: placeDetails.latitude,
      });
      setDisplayAddress(
        `${placeDetails.prediction.structured_formatting.main_text} \n${
          placeDetails.prediction.structured_formatting.secondary_text
        } ${placeDetails.zip || ''}`
      );
      setShouldSaveAddress(false);
    }
  }, [shouldSaveAddress, placeDetails]);

  useEffect(() => {
    if (displayAddress) {
      setShowAddressSearch(false);
    }
  }, []);

  const handleSuiteChange = (value: string) => {
    setSeekerAddress({
      ...seekerAddress,
      addressLine2: value,
    });
  };

  const addressSearchMode = showAddressSearch || shouldSaveAddress;

  useEffect(() => {
    if (addressSearchMode && !predictions) {
      logIbAddressErrorEvent(notFoundText, pageName);
    }
  }, [addressSearchMode, predictions]);
  useEffect(() => {
    if (!addressSearchMode && !isValidAddress) {
      logIbAddressErrorEvent(invalidAddressText, pageName);
    }
  }, [addressSearchMode, isValidAddress]);

  return (
    <Box>
      {addressSearchMode ? (
        <Box className={classes.addressSearchContainer}>
          <Icon24UtilitySearch className={classes.loopIcon} />
          <GoogleMaps.InputSearch label={label} onFocus={handleFocus} onBlur={handleBlur} />
          {predictions ? (
            predictions.map((prediction) => (
              <GoogleMaps.Prediction
                {...prediction}
                key={prediction.place_id}
                onClick={handlePredictionClick}
              />
            ))
          ) : (
            <Box className={classes.notFoundContainer}>
              <Typography variant="h2">{notFoundText}</Typography>
              <Typography variant="body1">{notFoundDescription}</Typography>
            </Box>
          )}
        </Box>
      ) : (
        <Box>
          <Box className={classes.locationContainer}>
            <Box className={classes.locationText}>
              <Typography>{displayAddress}</Typography>
            </Box>
            <Box>
              <Button
                color="secondary"
                variant="outlined"
                size="small"
                className={classes.editLocationButton}
                onClick={handleEditClick}>
                Edit
              </Button>
            </Box>
          </Box>
          {!isValidAddress && (
            <Box mt={2}>
              <Typography variant="body2" color="error">
                {invalidAddressText}
              </Typography>
            </Box>
          )}
          <Box className={classes.suiteContainer}>
            <InlineTextField
              id="aptOrSuite"
              InputLabelProps={{ shrink: false }}
              name="aptOrSuite"
              label={suiteLabelText}
              value={seekerAddress.addressLine2}
              onChange={(e) => handleSuiteChange(e.target.value)}
            />
          </Box>
        </Box>
      )}
    </Box>
  );
};

const AddressSearchWrapper = (props: AddressSearchProps) => {
  return (
    <GoogleMaps.Provider>
      <AddressSearch {...props} />
    </GoogleMaps.Provider>
  );
};

export default AddressSearchWrapper;
