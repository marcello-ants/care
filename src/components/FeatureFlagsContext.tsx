import { createContext, useContext, ReactNode, useState } from 'react';
import { LDEvaluationDetail } from 'launchdarkly-node-server-sdk';

interface FeatureFlagsProviderProps {
  children: ReactNode;
  flags: FeatureFlags;
}

export type featureFlagsState = {
  flags: FeatureFlags;
};

export type FeatureFlags = Record<string, LDEvaluationDetail | undefined>;

const initialState: featureFlagsState = {
  flags: {},
};

const FeatureFlagsContext = createContext<featureFlagsState>(initialState);

export const FeatureFlagsProvider = ({ children, flags }: FeatureFlagsProviderProps) => {
  const [state] = useState({ flags });
  return <FeatureFlagsContext.Provider value={state}>{children}</FeatureFlagsContext.Provider>;
};

export const useFeatureFlags = () => useContext(FeatureFlagsContext);
