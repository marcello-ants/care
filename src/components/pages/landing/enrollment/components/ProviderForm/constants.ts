export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
