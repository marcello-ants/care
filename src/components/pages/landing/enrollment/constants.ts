export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
