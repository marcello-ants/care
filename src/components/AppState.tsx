import { useRouter } from 'next/router';
import { createContext, useReducer, useContext, ReactNode, useEffect } from 'react';
import { debounce } from 'lodash-es';
import { configureScope, captureException, Severity } from '@sentry/nextjs';
import {
  initialAppState,
  rootReducer,
  isStateVersionCompatible,
  doesStateHaveDefaultKeys,
  validateState,
} from '@/state';
import patchAppState from '@/state/patch';
import { initialState as flowInitialState } from '@/state/flow';
import logger from '@/lib/clientLogger';
import { ClientLogger } from '@care/utils';
import getVertical from '@/utilities/verticalUtils';
import { SessionStorageHelper } from '@/utilities/SessionStorageHelper';
import { trackPageViewInSiftScience } from '@/utilities/siftScienceHelper';
import { AppAction, AppDispatch, AppState } from '@/types/app';
import { StateCheckingResponse } from '@/types/common';
import {
  LOCAL_STORAGE_STATE_KEY,
  CZEN_BASE_PATH,
  FLOWS,
  POST_A_JOB_ROUTES,
  CLIENT_FEATURE_FLAGS,
} from '@/constants';
import { useFeatureFlags } from '@/components/FeatureFlagsContext';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

type AppStateProviderProps = {
  children: ReactNode;
  initialStateOverride?: AppState;
  czenJSessionId?: string | undefined;
  referrerCookie?: string | undefined | null;
  userHasAccount?: boolean;
  memberUuid?: string;
};

function getStateFromStorageOrDefault(defaultState: AppState) {
  let stateFromStorageJson;
  const sessionStorageAvailable = SessionStorageHelper.isAvailable();

  if (sessionStorageAvailable) {
    stateFromStorageJson = SessionStorageHelper.getItem(LOCAL_STORAGE_STATE_KEY);
  } else {
    logger.warn({ event: 'sessionStorageUnavailable' });
  }

  if (stateFromStorageJson) {
    const stateFromStorage = JSON.parse(stateFromStorageJson) as AppState;
    try {
      const patchedState = patchAppState(stateFromStorage, defaultState);

      if (isStateVersionCompatible(patchedState) && doesStateHaveDefaultKeys(patchedState)) {
        return patchedState;
      }
      logger.warn({ event: 'invalidStateFromLocalStorage', state: stateFromStorageJson });
    } catch (e) {
      captureException(e, { level: Severity.Warning });
    }
  }
  return defaultState;
}

function persistState(state: AppState) {
  SessionStorageHelper.setItem(LOCAL_STORAGE_STATE_KEY, JSON.stringify(state));
}

const debouncePersistState = debounce(persistState, 500, {
  leading: true,
  trailing: true,
  maxWait: 1000,
});

function reducerWrapper(state: AppState, action: AppAction) {
  const nextState = rootReducer(state, action);
  debouncePersistState(nextState);
  return nextState;
}

export function redirectIfInvalidState(state: AppState, pathname: string): StateCheckingResponse {
  if (
    typeof Intl === 'undefined' ||
    typeof Intl.NumberFormat === 'undefined' ||
    typeof Intl.DateTimeFormat === 'undefined'
  ) {
    // TODO: Remove this with SC-214
    logger.error({ event: 'IntlUnavailable', source: pathname, invalidState: state });
    return {
      redirectCalled: true,
      readableMessage: 'Safari 9 and lower is not currently supported',
    };
  }

  const isStateValid = validateState(state, pathname);
  const redirect = !isStateValid;

  if (redirect) {
    SessionStorageHelper.removeItem(LOCAL_STORAGE_STATE_KEY);

    window.location.assign(CZEN_BASE_PATH);
  }

  return { redirectCalled: redirect };
}

export function getFlowName(pathname: string, oldFlowName?: string): string {
  let newFlowName = Object.values(FLOWS).find((flow) => pathname.includes(flow.path))?.name;

  if (!newFlowName) {
    // TODO: this shouldn't be needed after SC-123 is implemented
    const isPostAJob = Object.values(POST_A_JOB_ROUTES).some((route) => pathname.includes(route));
    if (isPostAJob) {
      newFlowName = 'SEEKER_POST_A_JOB';
    }
  }

  // Don't allow shifting from in-facility to seeker flow name
  if (newFlowName === FLOWS.SEEKER.name && oldFlowName === FLOWS.SEEKER_IN_FACILITY.name) {
    return FLOWS.SEEKER_IN_FACILITY.name;
  }

  return newFlowName ?? '';
}

function useFlowDetection(dispatch: AppDispatch, oldFlowName: string) {
  const { pathname } = useRouter();
  const newFlowName = getFlowName(pathname, oldFlowName);
  const vertical = getVertical();
  ClientLogger.setLoggerStatics({ flow: newFlowName, vertical });

  useEffect(() => {
    dispatch({ type: 'setFlowName', flowName: newFlowName });

    configureScope((scope) => {
      scope.setTags({
        flow: newFlowName,
        vertical,
      });
    });
  }, [newFlowName]);
}

const AppStateContext = createContext<AppState>(initialAppState);
const AppDispatchContext = createContext<AppDispatch>(() =>
  // eslint-disable-next-line no-console
  console.warn(
    'Calling default dispatch is a no-op! You are probably consuming app dispatch in a component above AppStateProvider'
  )
);

export const AppStateProvider = ({
  children,
  initialStateOverride,
  czenJSessionId,
  referrerCookie,
  userHasAccount,
  memberUuid,
}: AppStateProviderProps) => {
  const hydratedInitialAppState = {
    ...initialAppState,
    flow: {
      ...flowInitialState,
      czenJSessionId,
      referrerCookie,
      userHasAccount,
      memberUuid,
    },
  };
  const featureFlags = useFeatureFlags();
  const sessionStorageRetrievalSuccessEvaluation =
    featureFlags?.flags[CLIENT_FEATURE_FLAGS.SESSION_STORAGE_RETRIEVAL_SUCCESS];
  const shouldMeasureSessionStorageRetrievalSuccess =
    sessionStorageRetrievalSuccessEvaluation?.variationIndex === 1;

  const router = useRouter();

  const logSessionStorageError = () => {
    logger.info({ event: 'sessionStorageStateUnavailable', path: router.pathname });
    AnalyticsHelper.logEvent({
      name: 'sessionStorageStateUnavailable',
    });
  };
  const launchDarklyWrappedGetStateFromStorageOrDefault = (defaultState: AppState) => {
    if (shouldMeasureSessionStorageRetrievalSuccess) {
      try {
        const sessionStorageState = SessionStorageHelper.getItem(LOCAL_STORAGE_STATE_KEY);
        if (sessionStorageState && sessionStorageState.length > 0) {
          const stateFromStorage = JSON.parse(sessionStorageState);
          if (stateFromStorage) {
            logger.info({ event: 'sessionStorageStateAvailable', path: router.pathname });
          } else {
            logSessionStorageError();
          }
        } else {
          logSessionStorageError();
        }
      } catch (_e) {
        logSessionStorageError();
      }
    }
    return getStateFromStorageOrDefault(defaultState);
  };

  const [state, dispatch] = useReducer(
    reducerWrapper,
    initialStateOverride ?? hydratedInitialAppState,
    launchDarklyWrappedGetStateFromStorageOrDefault
  );
  useFlowDetection(dispatch, state.flow.flowName);
  const routePath = router.asPath?.split('?')[0];
  const stateCheckResult = redirectIfInvalidState(state, routePath);
  if (stateCheckResult.redirectCalled) {
    if (stateCheckResult.readableMessage) {
      return (
        <>
          <p>Redirecting to Member Home Page...</p>
          <p>{stateCheckResult.readableMessage}</p>
        </>
      );
    }
    return <></>;
  }

  useEffect(() => {
    trackPageViewInSiftScience(state);
  }, [routePath]);

  return (
    <AppDispatchContext.Provider value={dispatch}>
      <AppStateContext.Provider value={state}>{children}</AppStateContext.Provider>
    </AppDispatchContext.Provider>
  );
};

AppStateProvider.defaultProps = {
  initialStateOverride: undefined,
  czenJSessionId: undefined,
  referrerCookie: undefined,
  userHasAccount: false,
  memberUuid: undefined,
};

export const useAppState = () => useContext(AppStateContext);
export const useAppDispatch = () => useContext(AppDispatchContext);

export const useSeekerState = () => useContext(AppStateContext).seeker;
export const useProviderState = () => useContext(AppStateContext).provider;
export const useProviderCCState = () => useContext(AppStateContext).providerCC;
export const useFlowState = () => useContext(AppStateContext).flow;
export const useSeekerCCState = () => useContext(AppStateContext).seekerCC;
export const useSeekerHKState = () => useContext(AppStateContext).seekerHK;
export const useSeekerPCState = () => useContext(AppStateContext).seekerPC;
export const useSeekerTUState = () => useContext(AppStateContext).seekerTU;
export const useLtcgState = () => useContext(AppStateContext).ltcg;
