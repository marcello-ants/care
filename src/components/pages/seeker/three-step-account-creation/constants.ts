import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
