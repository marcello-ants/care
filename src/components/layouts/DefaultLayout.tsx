import { Container, makeStyles } from '@material-ui/core';
import { ILayoutProps } from '@/types/layout';
import { useFlowState } from '@/components/AppState';
import { FLOWS } from '@/constants';

const useStyles = makeStyles((theme) => ({
  root: {
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    paddingLeft: ({ isWideSpacingLayout }: { isWideSpacingLayout: boolean }) =>
      isWideSpacingLayout ? theme.spacing(3) : theme.spacing(2),
    paddingRight: ({ isWideSpacingLayout }: { isWideSpacingLayout: boolean }) =>
      isWideSpacingLayout ? theme.spacing(3) : theme.spacing(2),
    overflowX: 'hidden',
    boxSizing: 'border-box',
    flex: '1 0 auto',
    [theme.breakpoints.up('md')]: {
      paddingTop: theme.spacing(6),
      paddingBottom: theme.spacing(6),
    },
  },
  content: {
    margin: 'auto',
    maxWidth: '410px',
  },
}));

function DefaultLayout({ children }: ILayoutProps) {
  const { flowName } = useFlowState();

  /* istanbul ignore next */
  const isWideSpacingLayout =
    flowName === FLOWS.SEEKER_IN_FACILITY.name ||
    flowName === FLOWS.SEEKER.name ||
    flowName === FLOWS.LTCG.name;

  const classes = useStyles({ isWideSpacingLayout });

  return (
    <Container className={classes.root}>
      <div className={classes.content}>{children}</div>
    </Container>
  );
}

export default DefaultLayout;
