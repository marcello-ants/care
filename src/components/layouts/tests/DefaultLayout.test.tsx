import { render } from '@testing-library/react';
import { useRouter } from 'next/router';
import { ThemeProvider } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { CLIENT_FEATURE_FLAGS, CC_REBRANDED_PATHS } from '@/constants';
import DefaultLayout from '../DefaultLayout';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

let mockRouter: any | null = null;

const renderComponent = (ldFlags: FeatureFlags = {}) => {
  return render(
    <FeatureFlagsProvider flags={ldFlags}>
      <ThemeProvider theme={theme}>
        <DefaultLayout>test</DefaultLayout>
      </ThemeProvider>
    </FeatureFlagsProvider>
  );
};

beforeEach(() => {
  mockRouter = {
    asPath: '',
    pathname: CC_REBRANDED_PATHS[0],
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
});

describe('Default Layout component', () => {
  it('matches snapshot', () => {
    const view = renderComponent();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('matches rebranded snapshot', () => {
    const ldFlags: FeatureFlags = {
      [CLIENT_FEATURE_FLAGS.SEEKER_CC_REBRANDED_ENROLLMENT]: {
        reason: { kind: '' },
        value: 2,
        variationIndex: 2,
      },
    };

    const view = renderComponent(ldFlags);

    expect(view.asFragment()).toMatchSnapshot();
  });
});
