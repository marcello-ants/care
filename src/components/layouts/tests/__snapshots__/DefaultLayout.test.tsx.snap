// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Layout component matches rebranded snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-4 makeStyles-root-6 MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-content-5"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default Layout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-1 makeStyles-root-3 MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-content-2"
    >
      test
    </div>
  </div>
</DocumentFragment>
`;
